code/game/turfs/simulated/walls.dm

```diff
@@ -370,19 +370,36 @@
 		return TRUE
 	return FALSE
 
-/turf/simulated/wall/proc/try_decon(obj/item/I, mob/user, params)
+/turf/simulated/wall/proc/try_decon(obj/item/I, mob/living/user, params)
 	if(istype(I, /obj/item/gun/energy/plasmacutter))
 		to_chat(user, "<span class='notice'>You begin slicing through the outer plating.</span>")
 		playsound(src, I.usesound, 100, 1)
-
-		if(do_after(user, istype(sheet_type, /obj/item/stack/sheet/mineral/diamond) ? 120 * I.toolspeed : 60 * I.toolspeed, target = src))
-			to_chat(user, "<span class='notice'>You remove the outer plating.</span>")
-			dismantle_wall()
-			visible_message("<span class='warning'>[user] slices apart [src]!</span>", "<span class='warning'>You hear metal being sliced apart.</span>")
+		if(do_after(user, I.toolspeed * 6 SECONDS, target = src))
+			cut_down(I, user)
 			return TRUE
-
+	if(istype(I, /obj/item/melee/energy))
+		var/obj/item/melee/energy/O = I
+		if(O.can_cut_walls && O.active)
+			to_chat(user, "<span class='notice'>You begin slicing through the outer plating.</span>")
+			playsound(src, 'sound/items/welder.ogg', 100, TRUE)
+			if(do_after(user, O.toolspeed * 10 SECONDS, target = src))
+				cut_down(I, user)
+				return TRUE
+	if(istype(I, /obj/item/twohanded/dualsaber))
+		var/obj/item/twohanded/dualsaber/O = I
+		if(O.wielded && !O.toy)
+			to_chat(user, "<span class='notice'>You begin slicing through the outer plating.</span>")
+			playsound(src, 'sound/items/welder.ogg', 100, 1)
+			if(do_after(user, O.toolspeed * 10 SECONDS, target = src))
+				cut_down(I, user)
+				return TRUE
```

Comment:
This entire check is super snowflakey, but I cant think of anything better on the top of my head.

Reply:
+1, it might be worthwhile and not too out of scope to refactor the proc in a similar manner as:

```dm
var/slice = FALSE
if(istype(I, /obj/item/gun/energy/plasmacutter))
	slice = TRUE // Kinda interesting this doesn't check for power, but oh well
else if(istype(I, /obj/item/melee/energy))
	var/obj/item/melee/energy/O = I
	slice = O.can_cut_walls && O.active
else if(istype(I, /obj/item/twohanded/dualsaber))
	var/obj/item/twohanded/dualsaber/O = I
	slice = O.wielded && !O.toy
if(!slice)
	return FALSE

to_chat(user, "<span class='notice'>You begin slicing through the outer plating.</span>")
playsound(src, 'sound/items/welder.ogg', 100, 1)
if(do_after(user, O.toolspeed * 10 SECONDS, target = src))
	cut_down(I, user)
	return TRUE
return FALSE
```

Reply:
Done!